src/pybind/mgr/dashboard/module.py

```diff
@@ -275,6 +275,7 @@ class Module(MgrModule, CherryPyConfig):
                min=400, max=599),
         Option(name='redirect_resolve_ip_addr', type='bool', default=False),
         Option(name='cross_origin_url', type='str', default=''),
+        Option(name='sso_oauth2', type='bool', default=False),
```

Comment:
Following up with the idea of encapsulating functionally related code, why not:
```suggestion
        from auth import OAuth2
        ...
        MODULE_OPTIONS.extend(OAuth.module_options),
```
At some point in the future, following this pattern we will be creating an interface for new functionality.

Reply:
I get the idea and tried to follow it but, for some reason it was breaking, when using the local import like in the suggestion above I was getting:

```
ceph-node-00 ceph-mgr[2737]: mgr[py] Traceback (most recent call last):
                                               File "/usr/share/ceph/mgr/dashboard/__init__.py", line 60, in <module>
                                                 from .module import Module, StandbyModule  # noqa: F401
                                               File "/usr/share/ceph/mgr/dashboard/module.py", line 231, in <module>
                                                 class Module(MgrModule, CherryPyConfig):
                                               File "/usr/share/ceph/mgr/dashboard/module.py", line 279, in Module
                                                 from services.auth.oauth2 import OAuth2
                                             ModuleNotFoundError: No module named 'services'
                                             ```